src/lib/hooks/useLoadResizeHook.js

```diff
@@ -0,0 +1,34 @@
+import { useLayoutEffect } from 'react';
+import { getElementsPositionDifference } from '../services/elementPositionService';
+
+export const useLoadResize = (effect, dependencies, contentEl, viewportEl, wait) => {
+  let throttleTimeout = null;
+
+  const callBack = () => {
+    effect(getElementsPositionDifference(contentEl, viewportEl));
+    throttleTimeout = null;
```

Comment:
I am wondering, what's the reason to have `throttleTimeout` here? When we do not use `wait` than `throttleTimeout = null` is not necessary here, isn't it?

Reply:
If we don't wait (line 19), then no, `throttleTimeout` is set to `null` when it _already was_ `null`. But if we do wait (line 14), then `throttleTimeout` is reset back to `null` once it's successfully called.

Do you think it's better to distinct these cases in the callback function?

```js
throttleTimeout = setTimeout(callBack, wait, isDelayed);
```

```js
  const callBack = () => {
    effect(getElementsPositionDifference(contentEl, viewportEl));

    if (isDelayed) {
      throttleTimeout = null;
    }
  };
```

Reply:
It's there.